android-components/components/browser/state/src/main/java/mozilla/components/browser/state/search/SearchEngine.kt

```diff
@@ -27,6 +28,7 @@ data class SearchEngine(
     val type: Type,
     val resultUrls: List<String> = emptyList(),
     val suggestUrl: String? = null,
+    val isGeneral: Boolean = false,
```

Comment:
Wanted to ask about the direction to take in regards to user added search engines.
Because based on https://mozilla-hub.atlassian.net/browse/FNXV2-22038 only "general" search engines can be set as the default:
- should the default here be `true` or
- should we amend Fenix to go with "only general _or user added_ search engines can be set as the default"

cc @DreVla

Reply:
So, I thinking we will just set `isGeneral` when we call `createSearchEngine` in Fenix https://searchfox.org/mozilla-mobile/source/fenix/app/src/main/java/org/mozilla/fenix/settings/search/AddSearchEngineFragment.kt#180